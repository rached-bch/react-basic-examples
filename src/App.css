.memory {
  display: flex;
  flex-wrap: wrap;
  width: 900px;
  margin: auto;
  user-select: none;
}
.columns {
  padding: 10px;
  border: 1px solid #cccccc;
}
.user-list {
  width: 900px;
  margin: auto;
  background-color: #cccccc;
  padding: 5px;
}
.user-item {
  background-color: #999999;
  padding: 5px;
  border: 1px solid #000000;
}
