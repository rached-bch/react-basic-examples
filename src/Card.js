import React from "react";

const Card = ({ card, feedback, onClick }) => {
  return (
    <div className="card" onClick={onClick()}>
      H
    </div>
  );
};

export default Card;
