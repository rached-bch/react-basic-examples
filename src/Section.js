import React, { Component } from "react";
import PropTypes from "prop-types";

export default class Section extends Component {
  static defaultProps = {
    text: "Texte temporaire par d√©faut"
  };

  static propTypes = {
    text: PropTypes.string.isRequired
  };

  constructor(props) {
    super(props);
    this.textInput = props.text;
  }
  render() {
    return <div>{this.textInput}</div>;
  }
}
