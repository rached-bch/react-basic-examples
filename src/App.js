import React from "react";
import "./App.css";
import Card from "./Card";
import GuessCount from "./GuessCount";

function App() {
  console.log("Hello world");
  return (
    <div className="memory">
      <GuessCount guesses={0} />
      <Card card="😀" feedback="hidden" />
      <Card card="🎉" feedback="justMatched" />
      <Card card="💖" feedback="justMismatched" />
      <Card card="🎩" feedback="visible" />
      <Card card="🐶" feedback="hidden" />
      <Card card="🐱" feedback="justMatched" />
    </div>
  );
}
export default App;
