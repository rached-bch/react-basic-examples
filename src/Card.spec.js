import { expect } from "chai";
import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";

import App from "./App";
import GuessCount from "./GuessCount";
import Section from "./Section";
import Card from "./Card";

describe("<Card/>", () => {
  it("Card should trigger its `onClick` prop when clicked", () => {
    const onClick = sinon.spy();
    const wrapper = shallow(
      <Card card="😀" feedback="hidden" onClick={onClick} />
    );

    wrapper.simulate("click");
    expect(onClick).to.have.been.callCount(1);
  });

  it("should match its reference snapshot", () => {
    const onClick = sinon.spy();
    const wrapper = shallow(
      <Card card="😁" feedback="hidden" index={0} onClick={onClick} />
    );

    expect(wrapper).to.matchSnapshot();
  });
});
