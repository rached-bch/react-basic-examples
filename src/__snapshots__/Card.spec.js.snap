// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> should match its reference snapshot 1`] = `
<div
  className="card"
>
  H
</div>
`;
